lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7801,16 +7801,10 @@ getRefParentOrDiag(const clang::RecordDecl *decl, ASTContext &ctx,
     for (const auto *attr : refParentDecl->getAttrs()) {
       if (const auto swiftAttr = llvm::dyn_cast<clang::SwiftAttrAttr>(attr)) {
         const auto &attribute = swiftAttr->getAttribute();
-        llvm::SmallVector<ValueDecl *, 1> valueDecls;
-        if (attribute.starts_with(retainPrefix)) {
-          auto name = attribute.drop_front(retainPrefix.size()).str();
-          valueDecls = getValueDeclsForName(decl, ctx, name);
```

Comment:
This `getValueDeclsForName` call triggered the import that was responsible for the circular references.